lib/providers/storage.dart

```diff
@@ -0,0 +1,44 @@
+import 'package:concordium_wallet/entities/accepted_terms_and_conditions.dart';
+import 'package:concordium_wallet/state/network.dart';
+import 'package:hive_flutter/hive_flutter.dart';
+
+/// Service for interacting with [Hive].
+class StorageProvider {
+  final Box<AcceptedTermsAndConditions> _acceptedTermsAndConditionBox;
+
+  const StorageProvider._(this._acceptedTermsAndConditionBox);
+
+  static Future<StorageProvider> init() async {
+    await Hive.initFlutter();
+    _registerAdapters();
+    await _openBoxes();
+
+    return StorageProvider._(await Hive.openBox<AcceptedTermsAndConditions>(AcceptedTermsAndConditions.table));
+  }
+
+  /// Register all adapters needed for typed boxes.
+  static void _registerAdapters() {
+    Hive.registerAdapter(AcceptedTermsAndConditionsAdapter());
+  }
+
+  /// Opens all boxes asynchronously.
+  static Future<void> _openBoxes() async {
```

Comment:
Opening all boxes means that we will always load all persisted data into memory. We generally have very little data so this might be fine, but the best practise would be to only open boxes on demand (and then close them again when done using them).

Have you considered this?

Reply:
Yes, I initially considered using LazyBox to prevent everything from being fetched into memory, especially since we're dealing with relatively little data. 

However, upon further reflection after your comment, I agree that a combination of LazyBox and normal Box might be more appropriate. It's essential to differentiate not just based on the data volume but also on how frequently it needs to be loaded. For instance, in the case of the terms and conditions, which aren't accessed frequently, using a LazyBox to keep it out of memory makes perfect sense. I've already made the necessary changes to implement the use of LazyBox.

Additionally, incorporating metrics into our app would be highly beneficial. It would provide us with insights into potential bottlenecks and allow us to optimize performance effectively.